generative/maisi/scripts/sample.py

```diff
@@ -0,0 +1,747 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from datetime import datetime
+import json
+
+
+import monai
+import torch
+from monai.transforms import Compose, SaveImage
+from monai.data import MetaTensor
+from monai.inferers import sliding_window_inference
+from monai.utils import set_determinism
+from tqdm import tqdm
+from generative.metrics import FIDMetric
+from generative.inferers import LatentDiffusionInferer
+from utils import binarize_labels, MapLabelValue, general_mask_generation_post_process, get_body_region_index_from_mask
+from find_masks import find_masks
+from augmentation import augmentation
+
+
+class ReconModel(torch.nn.Module):
+    def __init__(self, autoencoder, scale_factor):
+        super().__init__()
+        self.autoencoder = autoencoder
+        self.scale_factor = scale_factor
+
+    def forward(self, z):
+        recon_pt_nda = self.autoencoder.decode_stage_2_outputs(z / self.scale_factor)
+        return recon_pt_nda
+
+
+def ldm_conditional_sample_one_mask(
+    autoencoder,
+    difusion_unet,
+    noise_scheduler,
+    scale_factor,
+    anatomy_size,
+    device,
+    latent_shape,
+    label_dict_remap_json,
+    num_inference_steps=1000,
+):
+    with torch.no_grad():
+        with torch.cuda.amp.autocast():
+            # Generate random noise
+            latents = (
```

Comment:
Maybe we can have fewer line breaks here, or just have a function
```
def generate_initial_noise(latent_shape, device):
    """Generate initial random noise for the diffusion process."""
    return torch.randn([1,] + list(latent_shape)).half().to(device)
```

Reply:
Thanks, done